pandas/_libs/lib.pyx

```diff
@@ -2315,3 +2312,199 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
+                                                 bint keep_trivial_numbers):
+    """
+    Concatenates `rows_count` elements from each `col_count` numpy arrays
+    in `date_cols` into strings.
+
+    Note
+    ----
+    This function speeds up concatenation for numpy arrays.
+    You also can use `_concat_date_cols_sequence` function.
+
+    Parameters
+    ----------
+    date_cols : tuple of numpy arrays
+    rows_count : Py_ssize_t
+    col_count : Py_ssize_t
+    keep_trivial_numbers : bool, default False
+        if True and len(date_cols) == 1, then
+        conversion (to string from integer/float zero) is not performed
+
+    Returns
+    -------
+    arr_of_rows : ndarray (dtype=object)
+    """
+    cdef:
+        Py_ssize_t col_idx, row_idx
+        list list_to_join
+        cnp.ndarray[object] iters
+        object[::1] iters_view
+        flatiter it
+        cnp.ndarray[object] result
+        object[:] result_view
+
+    result = np.zeros(rows_count, dtype=object)
+    result_view = result
+
+    if col_count == 1:
+        array = date_cols[0]
+        it = <flatiter>PyArray_IterNew(array)
+        for row_idx in range(rows_count):
+            item = PyArray_GETITEM(array, PyArray_ITER_DATA(it))
+            result_view[row_idx] = convert_to_unicode(item,
+                                                      keep_trivial_numbers)
+            PyArray_ITER_NEXT(it)
+    else:
+        # create fixed size list - more effecient memory allocation
+        list_to_join = [None] * col_count
+        iters = np.zeros(col_count, dtype=object)
+        # create memoryview of iters ndarray, that will contain some
+        # flatiter's for each array in `date_cols` - more effecient indexing
+        iters_view = iters
+        for col_idx, array in enumerate(date_cols):
+            iters_view[col_idx] = PyArray_IterNew(array)
+        # array elements that are on the same line are converted to one string
+        for row_idx in range(rows_count):
+            for col_idx, array in enumerate(date_cols):
+                # this cast is needed, because we did not find a way
+                # to efficiently store `flatiter` type objects in ndarray
+                it = <flatiter>iters_view[col_idx]
+                item = PyArray_GETITEM(array, PyArray_ITER_DATA(it))
+                list_to_join[col_idx] = convert_to_unicode(item, False)
+                PyArray_ITER_NEXT(it)
+            result_view[row_idx] = PyUnicode_Join(' ', list_to_join)
+
+    return result
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef cnp.ndarray[object] _concat_date_cols_sequence(tuple date_cols,
+                                                    Py_ssize_t rows_count,
+                                                    Py_ssize_t col_count,
+                                                    bint keep_trivial_numbers):
+    """
+    Concatenates `rows_count` elements from each `col_count` sequences
```

Comment:
use a single Summary line, then you can put an extended line or lines if you need after

Reply:
done